include/event_manager_priv.h

```diff
@@ -101,6 +102,29 @@ extern "C" {
 	}
 
 
+/* Macro generates a function of name new_ename where ename is provided as
+ * an argument. Allocator function is used to create an event of the given
+ * ename type.
+ */
+#define _EVENT_ALLOCATOR_DYNDATA_FN(ename)				\
+	static inline struct ename *_CONCAT(new_, ename)(size_t size)	\
+	{								\
+		struct ename *event = k_malloc(sizeof(*event) + size);	\
+		static_assert((offsetof(struct ename, dyndata) +	\
+			       sizeof(event->dyndata.size)) ==		\
+			      sizeof(*event), "");			\
```

Comment:
Maybe: `static_assert(offsetof(struct ename, header) == 0, "");` to ensure header is first datafield

Reply:
Fixed